pkg/runtime/interfaces/application_configuration.go

```diff
@@ -129,7 +129,9 @@ type NotificationsConfig struct {
 	// scheme is used.
 	Type string `json:"type"`
 	// Some cloud providers require a region to be set.
-	Region                       string                       `json:"region"`
+	Region string `json:"region"`
```

Comment:
Ideally we should have 2 different structures.
AWS {
Region:
}

GCP {
projectID
}

etc

Reply:
Yeah that makes sense. Unfortunately that would be a breaking change. How do we handle this kind of change?

Reply:
Also there are a few other places having `Region` defined, should we change them all? It does feel out of scope of this PR though.

Reply:
Played with this idea a bit. PTAL https://github.com/honnix/flyteadmin/pull/1

Reply:
Ping @kumare3

Reply:
It is out of the scope of this PR. But, if you can create a follow up issue we are golden :)